contrib/fonttools/showttf.c

```diff
@@ -5542,7 +5542,7 @@ static void readcffprivate(FILE *ttf, struct topdicts *td, char **strings, int s
 	fseek(ttf,td->cff_start+td->private_offset+td->subrsoff,SEEK_SET);
 	readcffsubrs(ttf,td,&td->local_subrs, 1, name );
     }
-    if (name != "<Nameless>") free(name);
+    if (name != NULL && *name != "<Nameless>") free(name);
```

Comment:
this makes no sense, `name` is a `char*`, and you're dereferencing it to compare against a const string?

Reply:
Thank you for your feedback.
I saw this:
   5421     name = td->fontname?td->fontname:
   5422                 td->sid_fontname?getsid(td->sid_fontname,strings,smax):"<Nameless>";
Re reading these lines, I'm not sure know what to do.

What should I put here?

Reply:
Sorry, I meant "I'm not sure now what to do"

Reply:
I should have caught this last time -- sorry. Too much scripting work in my recent history. 

Try putting `const char *nameless_str = "<Nameless>"` at the top and using `nameless_str` on both lines 5422 and 5545. That gives you proper pointer equality for the lifetime of the function.

Reply:
That should use `strcmp()`, that is how strings are compared in C, not by checking pointers for equality.

Reply:
That's true in general but would be a weird choice in this case. The "question of interest" is whether `name` points to memory that should be `free`d. A pointer comparison is usually the right kind of test for that. The problem in the initial fix is that the location of the "\<Nameless\>" string is unknown due to how it is originally assigned. 

@jtanx is of course right that just dynamically allocating a "\<Nameless\>" string is a fine solution.  I'm giving advice in the direction of @serval2412's chosen approach because it's not misguided and I figure the folks making these smaller tool-driven improvements move on to larger changes by learning this way.

Reply:
I see. What I have seen elsewhere in this code base is to set a boolean variable and only free when it is true, but this requires rewriting the `name` assignment at the top (which would be a good thing instead of the terrible multilevel use of ?: operator)